confidential-data-hub/hub/src/secret/mod.rs

```diff
@@ -58,6 +58,12 @@ impl Secret {
             return Err(SecretError::ParseFailed("malformed input sealed secret"));
         }
 
+        if sections[2].ends_with('=') {
+            return Err(SecretError::ParseFailed(
+                "base64 payload contains padding character '='. Sealed secrets must use URL_SAFE_NO_PAD encoding without padding",
+            ));
+        }
+
         let secret_json = b64
             .decode(sections[2])
             .map_err(|_| SecretError::ParseFailed("base64 decode Secret body"))?;
```

Comment:
Thanks for this patch @beraldoleal for user.

I noticed that we have an error message here. If the error message you'd got is this, is it ok to make here "base64 url safe(without '=' paddings) decode secret body failed" or something similar?

Reply:
Makes sense. pushed.